float64[] position
float64[] velocity

bool[] connected
bool[] limit_reached